File: leetcode/169e.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        counter = {}
        for num in nums:
            if num not in counter:
                counter[num] = 1
            else:
                counter[num] += 1
        max_count, max_el = 0, None
        for num, count in counter.items():
            if count >= max_count:
                max_count = count
                max_el = num
        return max_el

File: leetcode/169e.java
class Solution {
    public int majorityElement(int[] nums) {
        Map<Integer, Integer> foo = new HashMap<>();

        if (nums.length <= 0) {
            return 0;
        }

        for (int num : nums) {
            if (foo.containsKey(num)) {
                foo.put(num, foo.get(num) + 1);
            } else {
                foo.put(num, 1);
            }
        }

        int maxKey = -1;
        int maxCount = 0;
        for (int key : foo.keySet()) {
            int count = foo.get(key);
            if (count > maxCount) {
                maxKey = key;
                maxCount = count;
            }
        }

        return maxKey;
    }
}
